### PR TITLE
Split up redirection file

### LINKED ISSUE #27033: Break up redirection file
This issue concerns breaking up the [.openpublishing.redirection.json](https://github.com/dotnet/docs/blob/main/.openpublishing.redirection.json) file into multiple, smaller files. The additional redirection files can either live in the same directory as the main one, or in subdirectories. If they live in subdirectories, they don't need unique names.

Some tools might need to be updated if we break the file up, e.g.:

- https://github.com/dotnet/docs-tools/tree/main/actions/docs-verifier
- https://github.com/gewarren/cleanrepo

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #27033 

I did verify that a random redirect link from one of the new files works as expected: https://review.docs.microsoft.com/en-us/dotnet/azure/dotnet-howto-choose-migration?branch=pr-en-us-27444

### CODE REVIEW

**Comment by pkulikov** [comment]
@gewarren the merge conflict here is caused by the merged #27410 (at least) - I've added four redirection entries for the `csharp` folder

**Comment by gewarren** [comment]
I'll wait for approval before resolving any conflicts, to make sure I can merge it right away. This PR will have a tendency to accumulate merge conflicts.

**Review by BillWagner — APPROVED** [review]
This is a great step forward @gewarren 

Let's do it!

**Pull request merged**
